spring-integration-file/src/main/java/org/springframework/integration/file/remote/RemoteFileTemplate.java

```diff
@@ -420,7 +439,7 @@ public boolean get(final String remotePath, final InputStreamCallback callback)
 			throw new MessagingException("Failed to execute on session", e);
 		}
 		finally {
-			if (session != null) {
+			if (this.activeTemplateCallbacks.get() == 0 && session != null) {
```

Comment:
This isn't right - it might be another thread that's running in an invoke.

See `invokeScope` variable in rabbit template's doExecute